hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlPlanCacheTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl;
+
+import com.hazelcast.jet.sql.SqlTestSupport;
+import com.hazelcast.jet.sql.impl.connector.test.TestBatchSqlConnector;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SqlPlanCacheTest extends SqlTestSupport {
+
+    @BeforeClass
+    public static void setUpClass() {
+        initialize(2, null);
+    }
+
+    @Test
+    public void test_distributedInvalidation() {
+        TestBatchSqlConnector.create(instance().getSql(), "t", 1);
```

Comment:
We can call the `setStateCheckFrequency` to disable plan checking to ensure the plan is invalidated when the mapping is changed and not by the plan checker.